src/Tempest/Console/tests/TextBufferTest.php

```diff
@@ -0,0 +1,304 @@
+        $buffer->moveCursorToStartOfLine();
+        $this->assertSame($expected, $buffer->cursor);
+    }
+
+    #[Test]
+    public function test_move_cursor_to_end_of_line(): void
+    {
+        $buffer = new TextBuffer('Hello, world!');
+        $buffer->setCursorIndex(0);
+        $buffer->moveCursorToEndOfLine();
+        $this->assertSame(13, $buffer->cursor);
+    }
+
+    #[TestWith([0, 14])]
+    #[TestWith([5, 14])]
+    #[TestWith([14, 14])]
+    #[TestWith([15, 40])]
+    #[TestWith([20, 40])]
+    #[TestWith([40, 40])]
+    #[TestWith([41, 48])]
+    #[TestWith([45, 48])]
+    #[TestWith([48, 48])]
+    #[Test]
+    public function test_move_cursor_to_end_of_line_multiline(int $initial, int $expected): void
+    {
+        $buffer = new TextBuffer(<<<TXT
+        This is a line
+        This is a way longer line
+        Shorter
+        TXT);
+
+        $buffer->setCursorIndex($initial);
+        $buffer->moveCursorToEndOfLine();
+        $this->assertSame($expected, $buffer->cursor);
+    }
+
+    #[TestWith([0, 100, 13])]
+    #[TestWith([13, 1, 13])]
+    #[TestWith([0, -1, 0])]
+    #[TestWith([0, -100, 0])]
+    #[TestWith([0, 1, 1])]
+    #[TestWith([0, 13, 13])]
+    #[TestWith([4, -4, 0])]
+    #[Test]
+    public function test_move_cursor_x(int $initialCursor, int $offsetX, int $expectedPosition): void
+    {
+        $buffer = new TextBuffer('Hello, world!');
+
+        $buffer->cursor = $initialCursor;
+        $buffer->moveCursorX($offsetX);
+        $this->assertSame($expectedPosition, $buffer->cursor);
+    }
+
+    #[TestWith([0, 1, 15])]
+    #[TestWith([15, 1, 41])]
+    #[TestWith([41, 1, 41])]
+    #[TestWith([0, -1, 0])]
+    #[TestWith([15, -1, 0])]
+    #[TestWith([41, -1, 15])]
+    #[TestWith([14, 1, 29])]
+    #[TestWith([29, 1, 48])]
+    #[TestWith([40, 1, 48])]
+    #[TestWith([40, -1, 14])]
+    #[Test]
+    public function move_cursor_y(int $initialCursor, int $offsetY, int $expectedPosition): void
+    {
+        $buffer = new TextBuffer(<<<TXT
+        This is a line
+        This is a way longer line
+        Shorter
+        TXT);
+
+        $buffer->setCursorIndex($initialCursor);
+        $buffer->moveCursorY($offsetY);
+        $this->assertSame($expectedPosition, $buffer->cursor);
+    }
+
+    #[TestWith(['123', 0, [0, 0]])]
+    #[TestWith(["123\n456", 4, [0, 1]])]
+    #[TestWith(["123\n456", 5, [1, 1]])]
+    #[TestWith(["123\n\n456", 5, [0, 2]])]
+    #[TestWith(["different\nline\nlength", 9, [9, 0]])]
+    #[TestWith(["different\nline\nlength", 10, [0, 1]])]
+    #[TestWith(["different\nline\nlength", 11, [1, 1]])]
+    #[TestWith(["different\nline\nlength", 21, [6, 2]])]
+    public function test_relative_cursor_index(string $initialText, int $cursor, array $expectedPoint): void
+    {
+        $buffer = new TextBuffer($initialText);
+
+        $buffer->cursor = $cursor;
+        $point = $buffer->getRelativeCursorPosition();
+
+        $this->assertSame($expectedPoint[0], $point->x);
```

Comment:
Have you tried `assertStringEqualsStringIgnoringLineEndings` (there might be a typo in the method name, I write it from memory)

Reply:
In that case, the assertion checks for coordinates, so that's not the solution. The issue might be in the manual `\n` used to build strings in tests, but I suspect there might be other issues somewhere else as well, not just in the assertions